%!TEX root = ../thesis.tex
\chapter{Background}
\label{ch:background}

This thesis uses newly developed approaches from applied topology to study problems in evolutionary biology and genomics.
Here we supply sufficient background to motivate our approach.

\section{Biology}

In this section we present a basic introduction to molecular sequence data: what the data looks like, the processes by which it is generated, and the methods by which it is analyzed.
Particular attention is paid to modes of reticulate evolution.
Exposition for specific applications can be found in their respective individual chapters.

\subsection{Genes and Genomes}

The information required to express an organism's biological form and function is contained in the genome.
Physically, the genome is manifest as a sequence of nucleotides (DNA), at least one copy of which is packaged inside each cell of an organism.
Abstractly, the genome is represented as a linear sequence of characters defined over the alphabet $\{A,C,G,T\}$.\footnote{The linear representation can be misleading, as many organisms, primarily viruses and bacteria, have circular genomes.}
Contained in this sequence are subsequences representing genes, which code for the protein products that ultimately affect function.
Further embedded in the genome is a complex regulatory pattern of transcription factors controlling the expression of particular genes and directing cellular differentiation and development.

Following the central dogma of biology, DNA is transcribed into RNA, RNA is translated into amino acids, and amino acids are folded into proteins \cite{Crick:1970wb}.
Proteins comprise the functional unit of biology.

Beyond simply coding for function, the genome includes an imprint of the evolutionary history that gave rise to the organism.
By comparing the genomes of multiple organisms, inferences can be drawn about the evolutionary relationships among extant organisms as well as the processes that generated bserved diversity.
The field concerned with exploring these relationships is \emph{comparative genomics}.

\subsection{Evolutionary Processes}

Evolution describes the gradual change in phenotypes arising from random variation and subject to natural selection.
The processes giving rise to diversity can be classified into two types: clonal and reticulate.

\subsubsection{Clonal Evolution}

Clonal evolution, or vertical evolution, is a process of self-reproduction whereby genetic material is transferred directly from parent to offspring.
Population diversity is generated by stochastic mutation and maintained over multiple generations by random drift.

It is clonal evolution that Darwin had in mind when he described the idea of descent with modification, whereby a parent passes genomic information to an offspring subject to random drift.
Importantly, because there is always a direct parent--offspring relationship, clonal evolution will be consistent with a phylogenetic tree model.

\subsubsection{Reticulate Evolution}

Reticulate evolution, or horizontal evolution, refers to exchange or acquisition of genetic material via processes that do not reflect a direct parent--offspring relationship.
As we will see, these processes can make inferences about historical evolutionary relationships difficult.
Different types reticulate processes occur in different types of organisms (summarized in Table \ref{table:reticulation_processes}).

Viruses replicate by infecting a host cell and then using the host cell machinery and resources to produce multiple copies of viral genetic material.
The genetic material is then packaged into new virus particles which are shed off in order to infect new cells.
Reticulation can occur when two virus particles coinfect the same host cell.
During the replication process, genetic material can be exchanged in one of two ways: \emph{reassortment} or \emph{recombination} (the two processes are contrased in Figure~\ref{fig:viral_reticulation}).
Reassortment occurs in viruses whose genomes are segmented, such as influenza.
Segments are similar to chromosomes, such that a single virus particle will contain a single copy of each segment.
Reassortment occurs when coinfection results in packaging of segments taken from different virus particles.
The result viral progeny will then be a genetic mixture of segments from each parental strain.
Recombination, more common in nonsegmented viruses such as HIV, involves a break-rejoin mechanism during the replication process.
Here, an error in the polymerase during replication can result in an incomplete copy of the genome (a break).
At this point, several cellular processes involved in repair can be recruited to complete the replication process using a homologous region.
If coinfection has occurred, it is possible for these processes to initiate repair using material from a different parental strain.
The outcome will be novel genetic material that includes a crossover from one strain to another.
Break-rejoin crossover is a type of \emph{homologous recombination}.

\begin{figure}
\centering
\includegraphics[]{./fig/background/viral_reticulation.pdf}
\caption[Viral recombination and reassortment]{The two modes of viral reticulation. Coinfection of the same host cell can lead to either reassortment, in which whole viral segments are exchanged, or recombination, in which breakpoints can occur within segments. The former process is common in influenza, the latter in HIV. The end result, however, is a novel virus particle which shares genetic information from both parents.}
\label{fig:viral_reticulation}
\end{figure}

In bacteria and other prokaryotes, reticulate evolution can occur when foreign DNA from a donor is acquired by a target organism and integrated into its genome.
Three generic mechanisms have been identified, depending on the route by which foreign DNA is acquired \cite{Ochman:2000dr}:
\begin{enumerate}
\item \emph{Conjugation}. Direct cell-to-cell contact between donor and recipient resulting in transfer of plasmid.
\item \emph{Transformation}. Foreign DNA acquired via uptake from freely circulating DNA in the environment.
\item \emph{Transduction}. Virus-mediated transfer for foreign DNA from an infected donor cell.
\end{enumerate}
A visualization of these three mechanisms is shown in Figure~\ref{fig:bacterial_reticulation}.
Beause these mechanisms can often lead to the acqusition of novel sequences coding for genes not in the recipient organism, reticulate evolution in prokaryotes is often called \emph{horizontal gene transfer} or \emph{lateral gene transfer}.

\begin{figure}
\centering
\includegraphics[]{./fig/background/bacterial_reticulation.pdf}
\caption[Three modes of bacterial reticulation]{Three modes of viral reticulation. (1) Conjugation, in which direct cell-to-cell contact results in transfer of genetic material; (2) Transformation, in which foreign DNA is acquired via uptake from freely circulating DNA in the environment; and (3) Transduction, in which exchange of genetic material is mediated by a virus or phage particle.}
\label{fig:bacterial_reticulation}
\end{figure}

In eukaryotes, several reticulate processes have been identified.
We mention two such processes: hybrid speciation and meitotic recombination.
These two processes act at very different scales, but the outcome is the same: a unique offspring with genetic material drawing from both parents.

First, hybrid speciation refers to the cross-breeding of animals or plants of different species.
This mixing of genetic material can lead to the development of a third species with a phenotype distinct from both parents.
Hybrid speciation was originally believed to be a rare occurrence in nature and hybrid offspring to be infertile.
However, recent genomic data has demonstrated that hybrid speciation occurs quite frequently in plants \cite{Arnold:1996,Arnold:2007vq}.
Indeed, Mendel's early experiments in hybridization were themselves an artifically induced form of reticulate evolution.

Second, meiotic recombination refers to a specialized process for generating diversity that occurs in sexually-reproducing polyploid organisms, such as humans, during meiosis.
Meiosis is the process by which a single cell containing $n$ copies of each chromosome results in four distinct cells each with $n/2$ copies of each chromosome.
These special cells are called gametes.
Sexual reproduction consists of the fusion of two gametes during fertilization to form a zygote, which ultimately develops into a viable offspring.
Meiosis is a multistep process consisting of an initial round of DNA replication followed by two rounds of cell division.
Meiotic recombination occurs after the initial round of DNA replication and prior to cell division.
After DNA replication, there are two copies of each homologous chromosome that are joined at a centromere.
The two sets of chromosomes then pair with each other and exchange DNA through physical interactions known as crossovers.\footnote{These crossovers have been shown to not occur randomly, but rather at recombination hotspots regulated by binding motifs for by the PRDM9 protein. \kje{Cite Pablo Paper and a Molly paper}.}
This is another example of homologous recombination and results in new allelic patterns mixing genetic information from both parents.\footnote{Patterns of shared alleles define the concept of \emph{linkage}.}
After crossover occurs, two phases of cellular division result in gametes with $n/2$ copies of each chromosome.

At this point, one might wonder about sexual reproduction -- an offspring can be seen as a hybridization of genetic material donated from both mother and father.
On the one hand, the answer could be yes, particulare because the presence of meiotic recombination involves a shuffling of genomic material such that the chromosome each parent donates is a unique combination of alleles not previously present in the donor organism.
On the other hand, the answer could be no, because both mother and father donte a complete copy of the genome to the offspring.
Each copy can be considered as an independent transfer of genetic information defining both a matrilineal and a patrilineal line of inheritence.
Indeed, researchers in human population genetics generally distinguish between these two cases -- looking at genomic regions that do not recombine they define a matrilinearl common ancestor known as \emph{Mitochondrial Eve} and a patrilineal common ancestor known as \emph{Y-chromosomal Adam}.
Mathematically, the evolutionary relationships of a population of $N$ organisms with ploidy $n$ are often analyzed as a haploid population of size $nN$ with random mating.

\begin{tabularx}{\textwidth}{lll}
\toprule
Organism & Process & Description \\
\midrule
\multirow{2}{*}{Virus} & Reassortment & Exchange of discrete genomic segments \\
                       & Recombination & Intragenomic homologous crossover \\
\midrule
\multirow{3}{*}{Bacteria} & Transformation & Acqusition of foreign DNA in environment \\
                          & Transduction   & Viral-mediated exchange \\
                          & Conjugation    & Cell-to-cell contact and exchange \\
\midrule
\multirow{2}{*}{Eukaryotes} & Meiotic Recombination & Homologous crossover during meiosis \\
                            & Hybrid Speciation         & Fertilization across species boundaries \\
\bottomrule
\label{table:reticulation_processes}
\end{tabularx}

The presence of reticulate processes in a set of organisms can be most clearly identified by comparing phylogenetic relationships built from different genomic segments.
A general practice is to construct the set of \emph{gene trees} which reflect ancestral branching patterns at specific loci.
If a reticulate event has occurred, it implies that the branching patterns of different genes will not agree.
A subfield of comparative genomics is concerned with building \emph{species trees} from sets of gene trees.

However, in the case where there is substantial disagreement among gene trees, the very notion of a species tree may be flawed.
Traditionally, evolutionary biology has concerned itself with characterizing relationships in light of vertical evolution alone.
However, increasing evidence \kje{(what evidence?)} has pointed to the important role played by horizontal evolution, particularly in prokaryotic evolution.
\kje{[Further citations and exposition of Doolittle, Koonin, and Gogarten.]}

\subsection{Mathematical Models of Evolution}

Mathematical population genetics is concerned with properties of populations as they are subject to evolutionary forces over long time scales.
These forces include natural selection, genetic drift, mutation, and recombination.
Historically the input data for population genetics models was comparative studies of allele frequencies across populations.
These studies have primarily been replaced by large-scale genomic surveys which have provided unprecedented insight into ancient population structure and historical migrations.


These models allow two things: 1) simulate genomic data under realistic processes and 2) build statistical models to estimate biological parameters from real data.

\subsubsection{The Wright-Fisher Model}

The Wright-Fischer model is a forward time simulation of an evolving population.
In the simplest case, the model describes neutral evolution of a constant population size with no structure and constant genome length.
The model proceeds in units of generations.
At each generation, a member of the population is an offspring of a randomly selected ancestor from the previous generation.
This offspring inherits its ancestors genomes, with mutations introduced at some base rate $\mu$.
A member of previous generation with no offspring will be considered extinct.

\subsubsection{The Coalescent Process}

The coalescent process is a stochastic model that generates the genealogy of individuals sampled from an evolving population \cite{Wakeley:2009}.
The genealogy is then used to simulate the genetic sequences of the sample.
This model is essential to many methods commonly used in population genetics.
Starting with a present-day sample of $n$ individuals, each individual's lineage is traced backward in time, towards a mutual common ancestor.
Two separate lineages collapse via a coalescence event, representing the sharing of an ancestor by the two lineages.
The stochastic process ends when all lineages of all sampled individuals collapse into a single common ancestor.
In this process, if the total (diploid) population size $N$ is sufficiently large, then the expected time before a coalescence event, in units of $2N$ generations, is approximately exponentially distributed:
\begin{equation}
P(T_{k}=t) \approx \binom{k}{2} e^{-\binom{k}{2} t},
\end{equation}
where $T_k$ is the time that it takes for $k$ individual lineages to collapse into $k-1$ lineages.

After generating a genealogy, the genetic sequences of the sample can be simulated by placing mutations on the individual branches of the lineage.
The number of mutations on each branch is Poisson-distributed with mean $\theta t / 2$, where $t$ is the branch length and $\theta$ is the population-scaled mutation rate.
In this model, the average \emph{genetic distance} between any two sampled individuals, defined by the number of mutations separating them, is $\theta$.

The coalescent with recombination is an extension of this model that allows different genetic loci to have different genealogies.
Looking backward in time, recombination is modeled as a splitting event, occurring at a rate determined by population-scaled recombination rate $\rho$, such that an individual has a different ancestor at different loci.
Evolutionary histories are no longer represented by a tree, but rather by an \emph{ancestral recombination graph}.
Recombination is the component of the model generating nontrivial topology by introducing deviations from a contractibile tree structure, and is the component which we would like to quantify.
Coalescent simulations were performed using \texttt{ms} \cite{Hudson:2002}.

\begin{figure}
\centering
  \subbottom[Wright-Fisher Model]{%
    \includegraphics[width=0.3\linewidth]{fig/background/wright-fisher_example.jpg}}
  \subbottom[Coalescent Model]{%
    \includegraphics[width=0.3\linewidth]{fig/background/coalescent_example.jpg}}
    \caption{Two models for simulating evolutionary data.}
\end{figure}

% \begin{subfigure}[b]{.1\textwidth}
% \includegraphics{fig/background/wright-fisher_example.jpg}
% \caption{Wright-Fisher Model}
% \end{subfigure}
% \begin{subfigure}[b]{.1\textwidth}
% \includegraphics{fig/background/coalescent_example.jpg}
% \caption{Coalescent Process}
% \end{subfigure}
% \end{figure}

\subsubsection{Metrics on Sequences}
\label{bg:subsubsec:metrics}

Evolutionary models require a notion of genetic divergence between sequences.
This leads to a discussion of the types of metrics that can be put on sets of sequences.\footnote{Before sequence can be compared, they must first be \emph{aligned}. A sequence alignment arranges the characters in a set of sequences into coluns such that characters sharing evolutionary identity are in the same column. Alignment is necessary because random insertions and deletions can change . The difficulty of performing an alignment will largely depend on the amount of evolutionary divergence in the set of sequences under consideration. Sequence alignment is a well studied topic but largely beyond the scope of this thesis, where we assume sufficient sequence similarity such that alignment can be performed with high confidence.}

The simplest model, and the one most commonly adopted in this thesis, is the Hamming metric, which simply counts the proporation of sites that differ between two aligned sequences.
For example, for two sequence $s_1=ACTTGAC$ and $s_2=AAGTGGC$, $d_{H}(s_1,s_2)=3/7$.
In general, the Hamming metric will underestimate divergences by not accounting for the possibility of back mutations.\footnote{A double mutation of the form $A\rightarrow C\rightarrow A$.}

More biologically motivated models will introduce corrections to account for assumptions about how sequences evolve.
These assumptions include the base frequency of each nucleotide as well as the substitution rates for each type of mutation.
The simplest of these models is the \emph{Jukes-Cantor model}.
This model defines an equal substitution rate $\mu$.
Inverting the probability of an alteration gives the divergence.
The Jukes-Cantor metric is defined as 
\begin{equation}
d_{JC}=-\frac{3}{4}\ln(1-\frac{4}{3}p),
\end{equation}
where $p$ is the proporation of sites that are different.

\subsection{Phylogenetic Methods}

A phylogenetic tree is a binary tree in which leaves are associated with particular species or taxa, and the branching pattern of the tree reflects diverging evolutionary relationships.
Branch lengths on the tree are associated with evolutionary divergence between sets of taxa.
Molecular phylogenetics refers to a large collection of methods for inferring branching patterns from aligned molecular sequence data.\footnote{See Felsenstein, \emph{Inferring Phylogenies} for a readable and thorough introduction to the field \cite{Felsenstein:2004ws}.}
In general, the problem of finding an optimal tree associated with sequence data is NP-complete \cite{Foulds:1982fn}, however several approximate methods have been developed.
The primary types of methods include maximum parsimony, distance-matrix methods, maximum likelihood (ML), and Bayesian inference.
Maximum parsimony attempts to find the phylogenetic tree that minimizes the number of evolutionary changes required to explain the observed sequences.
Distance-matrix methods first compute a matrix of pairwise distances between taxa and then find the tree that best approximates these distances.
ML and Bayesian methods use specific models of evolution to assign probability distributions over trees.
In this work we concentrate on distance-matrix methods because of their close connection with the finite metric spaces considered in applied topology.

\subsubsection{Distance-Matrix Methods}

Given a set of aligned molecular sequences, distance-matrix methods first compute the pairwise matrix of genetic distances using one of the metrics as described in Section~\ref{bg:subsubsec:metrics}.
Then, the binary tree that best approximates those distances is iteratively fit to this data.
This approach to phylogenetic inference were introduced by Cavalli-Sforza and Edwards in 1967 \cite{CavalliSforza:1967th} and Fitch and Margoliash in 1967 \cite{Fitch:1967we}.
The Fitch-Margoliash method uses a weighted least squares approach to tree-fitting, such that larger distances are weighted less, due to higher chances for random error.
Distance-matrix methods are popular for their high speed and scalability as well as high accuracy in most cases.

Currently, the most widely implemented distance-matrix method is neighbor-joining.\footnote{Neighbor joining was introduced by Saitou and Nei in 1987 \cite{Saitou:1987wo}.}
One particular reason neighbor-joining is popular is that under certain conditions, discussed below, it has been shown to exactly recover the correct tree.
The neighbor-joining algorithm is described in Algorithm~\ref{bg:alg:nj}.

\begin{algorithm}
    \KwData{$n \times n$ distance matrix $D$}
    \KwResult{Phylogenetic tree on $n$ leaves}
    \While{blah} {
        Compute $Q$ matrix\;
    }
    \caption{The Neighbor Joining Algorithm. Adapated from Wikipedia entry on Neighbor-Joining.}
    \label{bg:alg:nj}
\end{algorithm}

\subsubsection{Additive Metrics and the Four Point Condition}

Arbitrary distance matrices are unlikely to admit a tree representation.
Those that do are called \emph{additive metrics}, because they can be represented as an additive tree.
Additivity is the property that the distance between any two nodes will be equal to the sum of the branch lengths between them.
A distance matrix admits a tree representation if and only if it is additive.

There is a straight-forward condition that must be satisfied for additivity, known as the \emph{four point condition}.
For a distance matrix to admit a tree representation, 
\begin{equation}
d_{ij} + d_{kl} \leq \max\{d_{ik} + d_{jl} ,  d_{il} + d_{jk} \}
\end{equation}
for any four nodes $\{i,j,k,l\}$.
The condition implies that there is a labeling on the four nodes such that
\begin{equation}
d_{ij} + d_{kl} \leq d_{ik} + d_{jl} =  d_{il} + d_{jk}.
\end{equation}
A visual interpretation of this condition is shown in Figure~\ref{background:fig:four_point_condition}.

\begin{figure}
\centering
\includegraphics[]{./fig/background/four_point_condition.pdf}
\caption[The four point condition for additivity]{A visual interpretation of the four point condition for additivity. For any four leaves, there exists a labeling $\{i,j,k,l\}$ such that $d_{ij}+d_{kl}\leq d_{ik}+d_{il} = d_{il}+d_{jk}$. Of the three possible ways of arranging the sums of distances, two will involve traversing the internal branch, while one will involve only external branches.}
\label{background:fig:four_point_condition}
\end{figure}

Sequence data can fail to be additive for several reasons.
First, sequencing error.
Errors can introduce noise into the measured genetic distances.
Second, homoplasy.
A homoplasy occurs when the same mutation is introduced multiple times in a set of organisms.
The presence of homoplasy will underestimate genetic distance between taxa.
Third, reticulate evolution.
As described previously, in cases of reticulate evolution no tree will accurately describe the observed data.

\subsubsection{Number of Tree Topologies}

The number of unrooted bifurcating tree topologies with $L$ leaves is $(2L-5)!!$.\footnote{The double factorial is defined as $n!!=n(n-2)(n-4)\cdots$.}
This can be easily shown using induction.
For $L=3$, we have $\mathcal{T}(3)=1$ and $3$ branches.
To pass to $L=4$, we can add the fourth leaf to any of the $3$ branches, resulting in $3$ different topologies.
For $L=4$, we have $\mathcal{T}(4)=3$.
Every time we add a leaf, we add two branches -- one external and one internal.
For $L=n$, we have $\mathcal{T}(n)=(2n-5)!!$ and $2n-3$ branches.
For $L=n+1$, we can add the new external branch to any of the current $2n-3$ branches.
A rooted tree with $L$ leaves can be considered as an unrooted tree with $L+1$ leaves.
Therefore, the number of rooted bifurcating tree topologies with $L$ leaves is $(2L-3)!!$
As can be seen, the number of tree topologies explodes with the number of leaves.\footnote{It was observed by Walter Fitch that for more than 20 species there are more than Avogadro's number of topologies.}

\subsubsection{Space of Phylogenetic Trees \kje{(Needs Expansion)}}

A phylogenetic tree is characterized by its number of leaves, the particular tree topology, and the length of each branch.
Tree space refers to an abstract construction that represents each possible tree as a point in a geometric space.
Abstract studies of tree space were initiated by Billera, Holmes, and Vogtmann (BHV) in \cite{Billera:2001tv}.

In the BHV model, each point represents an unrooted binary tree with $L$ leaves and strictly positive branch lengths.
Number of interior edges $r=L-3$, a particular additive tree can be plotted as a point in the positive open orthant $(0,\infty)^{r}$.
A single orthant corresponds to a single tree topology.

The tree space construction allows one to define statistics, such as means and variance, on sets of trees.
This has been utilized 

\begin{figure}
\centering
\includegraphics[]{./fig/TreeSpace.pdf}
\caption[Tree Space Construction]{The construction of Tree Space. Each point in this space represented a particular unrooted tree topology with branch lengths.}
\label{background:fig:TreeSpace}
\end{figure}

\subsubsection{Phylogenetic Networks}

There are several existing methods for representing reticulate evolution.
Most of these methods generalize phylogenetic trees into \emph{phylogenetic networks}, which attempt to reconcile the presence of horizontal evolution in sequence data.
However, most simply present corrections to phylogenetic trees, which can fail in cases where horizontal evolution is pervasive, as in many prokaryote datasets.
Additionally, the resulting networks can be complex and difficult to interpret quantitatively.
This can make it difficult to distinguish between phylogenetic incompatibilities due to noisy sampling and due to true reticulations.
An example of a phylogenetic network using the split network approach is shown in Figure~\ref{background:fig:split_network}.

\kje{[Expand. See Morrison review..]}

\begin{figure}
\centering
\includegraphics[width=.8\textwidth]{fig/background/split_network.pdf}
\caption[Example of a Splits Network]{Example of a split network of genus Branchiopoda and outgroups. Computed using the Neighbor-Net algorithm. Phylogenetic incompatibilities are represented by conflicting splits. reprinted from BMC Evolutionary Biology 7:147 (2007).}
\label{background:fig:split_network}
\end{figure}

% NEXT: TDA SECTION. 

\section{Topological Data Analysis}
\label{bg:s:tda}

Topology is the branch of mathematics that formalizes our intuitive notions of shape.
More concretely, topology provides the methods to characterize the properties of objects and spaces that remain invariant under continuous deformation.
For example, squeezing a circle into an ellipse by compressing along one axis does not change the fact that the object encloses a single loop.
Or, as we saw in the introduction, the coffee mug can be continuously deformed into the donut.
Likewise, if we take a tree and change the lengths of its branches, the tree remains a tree.\footnote{It is important to draw a distinction between the notion of tree topology, in which the branch patterns determines the topology, and global topology, in which all trees are equivalent. While the former is more common in the phylogenetics community, here we consider the latter.}
In each of these examples, while the deformation has substantially altered \emph{local} properties of the space, on a \emph{global} level certain essential characteristics have remained unchanged.
From the perspective of topology, the spaces are considered identical.
The question then is how to formalize our idea of global shape in order to systematically reason about it.

Algebraic topology solves this problem by associating algebraic objects (an integer, for instance) that do not change under continuous deformation.
These objects may capture properties like the number of connected components, the number of loops, and the number of holes in an object, and represent \emph{topological invariants} of a space.
Two spaces can only be deformed into one other if they share the same invariants.
The circle and ellipse are identified as equivalent by the presence of one loop.
Neither can be deformed into a tree without introducing a cut, which would be a type of discontinuous deformation.
Using these invariants, powerful ideas from abstract algebra can then be used to manipulate and reason about shape.

While topology has traditionally developed through the study of abstract spaces, leading to very rich and beautiful constructions\footnote{For example, see the work of Thurstan on low-dimensional topology}, real data does not come in the form of perfect continuous spaces.
Recent effort over the past 15 years has focused on developing methods to apply topology to real world problems in science and engineering.
This work, collectively falling under the heading of \emph{topological data analysis} (TDA), has focused on efficient algorithms for computing topological invariants from finite, noisy data.
TDA now encompasses a range of efforts and can now be considered a branch of applied mathematics in its own right.
It has emerged from substantial interdisciplinary effort between mathematicians, computer scientists, and domain experts.

In practice, a typical workflow for applying TDA to real data is as follows.
Data comes in the form of a set of $n$ observations with $p$ attributes, where $p$ is often very large.
The data is assumed to be a finite sample from some more complex space, from which we wish to infer either some sort of global structure or underlying model.
The data is represented as a finite point cloud: a set of $n$ points in $p$ dimensions with an associated notion of distance.
The point cloud is then transformed into a topological space by associating different sets of points with each other.
The associations can be constructed in different ways -- for instance, one of the most common constructions associates points within a certain distance $d$ from one another.
Computational approaches are then used to measure informative topological properties from the space.

In this thesis, we use methods from TDA to study problems in evolutionary biology and genomics.
Our data is typically aligned genomic sequences from sets of related organisms.
If our sequences are each of length $l$, then we can imagine our data as points in an $l$-dimensional sequence space.
A sequence metric, such as the Hamming metric, measures distance.

The two main methods from TDA that we employ are \emph{persistent homology} and \emph{mapper}.
Persistent homology provides a way to efficiently compute the topological invariants of a space across multiple scales, while mapper provides an approach for a condensed representation and visualization.
In this section, we provide an overview and discussion of these two methods from the perspective of an end-user.
We treat each method as a pipeline for transforming from raw data to a topological summary.
While the mathematical literature on these methods is extremely deep, our goal is to explain things in sufficient detail for a wide audience to grasp the main ideas.
As such, we will define technical terms where necessary, but focus primarily on developing a working understanding of the main aspects of each method as they are used in practice.
The primary concept we require is \emph{homology}, a particular way in which topological invariants can be assigned to spaces.
We include a more thorough review of the requisite mathematical concepts leading to a definition of simplicial homology in Appendix XX.

The following sections draw on several excellent reviews of TDA, including \cite{Carlsson:2009a}, \cite{Edelsbrunner:2010}, and \cite{Ghrist:2008}.

\subsection{Persistent Homology}

% insert Persistent Homology Description Here.

\subsection{Mapper}

% insert Mapper description here.

\section{Applying TDA to Molecular Sequence Data}

Aligned molecular sequence data can be naturally viewed as a point cloud in a high-dimensional space, which we loosely call \emph{sequence space}.
The particular structure of sequence space will be determined by the length, $L$, of the aligned sequences, and the alphabet, $Q$, over which the sequences are defined.
A typical sequence alphabet will be either nucleotides or amino acids.
The dimension of the space is determined by $L$.
Sequence space will therefore consist of the $||Q||^L$ possible sequences.
Together with any of the standard genetic distance measures, this forms a finite metric space.

The process of evolution can be seen as an exploration of sequence space.
An individual genotype can be assigned a fitness, $w$, which will describe the probability of reproductive success in a particular environment.
Clonal evolution is the process of smoothly moving through sequence space, while reticulate evolution is the process of making discontinuous jumps through the space.

Our data generally consists of a subset of points sampled from sequence space.
These points reflect a particular evolutionary history.
As more data is acquired, areas of sequence space will become more densely sampled and the ability to reconstruct that evolutionary history will become more feasible.
Given molecular sequence data, our program is to (1) encode the data as a finite metric space, (2) use tools from TDA to characterize the topology of the data, and (3) interpret the topology in an evolutionary context.
In particular, we apply persistent homology, and read phylogenetic information contained in the dataset off the resulting barcode diagram.
We make two preliminary remarks before considering a more complete example.

\subsection{Topology of Tree-like Metrics}

An important foundational point was demonstrated by G. Carlsson in \cite{Chan:2013}.
Recall that tree-like data will have an additive metric space, as described in Section~XX.
In that paper, it was proven that for additive metric spaces, then the Vietoris-Rips filtration of the data will consist of a nested set of acyclic complexes.
Consequently, the persistent homology of additive data will have nontrivial topology only in dimension zero.
Furthermore, while noise in the data an introduce small deviations from additivity, the theorem puts bounds on the size of the topological features that can arise in this manner.
These bounds rely on the Gromov-Hausdorff stability conditions described in Section~XX.

On the other hand, if the evolutionary history includes reticulate events that cannot be represented as a tree, these events will be captured as non-trivial higher dimensional homology in the barcode diagram, an idea which we develop below.
This theorem provides an important positive control in using TDA to characterize reticulate evolution.

\subsection{The Fundamental Unit of Reticulation}

In population genetics, there is a simple test for the presence of reticulate evolution in sequence data called the \emph{four-gamete test} \cite{Hudson:1985}.
The test assumes only an infinite-sites model, which states that for a sufficiently long genome, a particular residue can only ever undergo a single mutation.
Put another way, there is no multiple-mutation or back mutation.
The infinite-sites model has three consequences: first, one need only consider segregating sites, or nucleotide positions that have undergone a mutation.
Second, because a given position can mutate only once, it is sufficient to represent sequences as binary strings, where a $0$ indicates the unmutated state and $1$ the mutated state.
Third, for a given position we can arbitrarily assign the unmutated and mutated states.
The infinite-sites model is considered a reasonably good model for long genomes.

The four-gamete test identifies reticulate evolution by looking at pairs of segregating sites.
Given biallelic data, there are four possible haplotype patterns, or states, for a pair of segregating sites: $00$, $10$, $01$, or $11$.\footnote{These sites need not be adjacent.}
The statement of the four gamete test is this: in any given dataset, the simultaneous presence of all four haplotype states in any pair of segregating sites is incompatible with strictly clonal evolution, and indicates reticulate evolution.
To see this, assume state $00$ as the ancestor to states $10$ and $01$, which arise from two independent mutations.
Because of the no multipe-mutation assumption, it is not possible for either of these two states to then independently mutate into state $11$.
The only way for state $11$ to arise is via a reticulate event that brings together the left site from state $10$ and the right site from $01$.\footnote{It is entirely possible for the reticulate event to have had a reversed pattern of ancestry, in which case the reticulation would result in a state $00$ and would not be detectable from the sequence data.}
This process is illustrated in Figure~XXA.

Under a Hamming metric, the distance matrix for the set of four sequences $s_1=00$, $s_2=10$, $s_3=01$, and $s_4=11$ is
\begin{equation}
d = 
\begin{pmatrix}
0 & 1 & 1 & 2 \\
1 & 0 & 2 & 1 \\
1 & 2 & 0 & 1 \\
2 & 1 & 1 & 2 \\
\end{pmatrix}
\end{equation}
The Vietoris-Rips filtration of this space is shown in Figure~\ref{background:fig:four_gamete_test}C.
At $\epsilon=0$ the four sequences are disconnected.
At $\epsilon=1$, four edges are drawn, forming a loop.
At $\epsilon=2$, the space is completely connected and the loop is killed.
Persistent homology captures the presence of this loop as an $H_1$ feature in the interval $[1,2)$ (Figure~\ref{background:fig:four_gamete_test}D).
In this way, the reticulate event is associated with the presence of a nonzero $H_1$ bar.

We consider this example to be the minimal, or fundamental, unit of reticulation.
All more complicated patterns of reticulation can be seen as extensions of this example.

\begin{figure}
\centering
\includegraphics[width=\textwidth]{./fig/background/four_gamete_test.pdf}
\caption[Fundamental Unit of Reticulation]{The Fundamental Unit of Reticulation. (A) A set of four sequences. (B) An evolutionary genealogy including two mutations ($m_1$ and $m_2$) and a single reticulation ($r_1$). (C) Vietoris-Rips Filtration. (D) Barcode Diagram.}
\label{background:fig:four_gamete_test}
\end{figure}

\subsection{A Complete Example}

We illustrate a complete example of how TDA can capture horizontal evolution from complex populaton data in Figure~\ref{background:fig:tda_on_sequencedata}.
Consider the reticulate phylogeny (Figure~\ref{background:fig:tda_on_sequencedata}A): five genetic sequences sampled today (yellow circles) originate from a single common ancestor due to clonal evolution (solid blue lines tracing parent to offspring) and reticulate evolution (dotted red lines).
In Figure~\ref{background:fig:tda_on_sequencedata}B, these five samples are placed in the context of a larger dataset, where the data has been projected onto the plane using PCA.
Persistent homology is then applied to this larger sample.
In Figure~\ref{background:fig:tda_on_sequencedata}C we demonstrate the construction of a filtered simplicial complex, showing how the connectivity changes as the scale parameter $\epsilon$ is increased.
Finally, in Figure~\ref{background:fig:tda_on_sequencedata}D we see the resulting barcode diagram.
Using $H_0$ we can track the number of strains or subclades that persist, roughly corresponding to the tree-like component of the data.
The $H_1$ bar near spanning roughly $\epsilon=0.13$ to $\epsilon=0.16$ identifies the presence of a reticulate event. involving the five higlighted sequences.
The scale over which this bar persists represents the amount of evolutionary time separating the parents and the reticulate offspring.
Additionally, the persistence algorithm will return a generating basis for a particular homology group, which we can use to identify the particular mixtures of sequences involved a reticulation.
In this way, we can analyze both the scale and frequency of reticulation in genomic data sets.

We summarize the connection between genomic data and TDA in Table~\ref{table:dictionary}.

\begin{table}
\centering
\caption{Dictionary connecting algebraic topology and evolutionary biology}
\small
\rowcolors{2}{gray!25}{white}
\setlength{\aboverulesep}{0pt}
\setlength{\belowrulesep}{0pt}
\setlength{\extrarowheight}{.75ex}
\begin{tabularx}{\textwidth}{XX}
\toprule\rowcolor{gray!50}
Algebraic Topology                                         & Evolutionary Biology \\
\midrule
Filtration value $\epsilon$                                & Genetic distance (evolutionary scale) \\
0-dimensional Betti number at filtration value $\epsilon$  & Number of clusters at scale $\epsilon$ \\
Generators of 0-D homology                                 & A representative element of the cluster \\
Hierarchical relationship among generators of 0-D homology & Hierarchical clustering \\
1-D Betti number                                           & Lower bound on number of reticulate events \\
Generators of 1-D Homology                                 & Reticulate events \\
Generators of 2-D Homology                                 & Complex horizontal genomic exchange \\
Non-zero high-dimensional homology (“topological obstruction to phylogeny”) & No treelike phylogenetic representation exists \\
Number of higher-dimensional generators over a time interval (“irreducible cycle rate”) & Lower bound on recombination/reassortment rate \\
\bottomrule
\end{tabularx}
\label{table:dictionary}
\end{table}

\begin{figure}
\centering
\includegraphics[width=\textwidth]{./fig/tda_on_sequencedata.pdf}
\caption[Applying TDA to Molecular Sequence Data]{Applying persistent homology to genomic data. (A) An evolutionary genealogy including reticulation. (B) Data projected into 2-dimensions. (C) Construction of a filtered simplicial complex. (D) The resulting multiscale barcode diagram.}
\label{background:fig:tda_on_sequencedata}
\end{figure}

\subsection{The Space of Trees, Revisited}

In Section~XX, Tree Space was introduced as an abstract construction to systematically represent the set of all possible unrooted binary trees.
Tree Space was shown to be a subspace of the space of finite metrics that satisfy the additivity condition.
Real data will very rarely satisfy this condition, and one possible interpretation of phylogenetic reconstruction is finding the best projection onto tree space for arbitrary data.




The program we propose is very closely related to this interpretation.
As we move off tree space, we know there is no additive tree that can represent that data.
We would like a topological measure of deviation from additivity.
Finite data comes represented as a metric space.
On top of this metric space we can associate some topological invariants, as computed using methods from TDA.
We will interpret these invariants as a measure of deviation from tree-like additivity.
The further the data sits from tree space, the more prominent these invariants should be, and the more reticulation we will associate with the data.

Rather than attempt to characterize arbitrary data as a projection onto tree space, we will use methods of TDA to compute topological invariants.

An ambitious goal would be a complete characterization of the space of finite metrics in terms of their topological invariants -- this is likely difficult.
Our updated picture looks as Figure~XX.

Note that this is an abstract construction and for data sets with an appreciable number of leaf nodes it will be difficult to directly visualize their position.

As data moves off tree space
Our hope is that the topological tools we develop here are sensitive enough to capture these deviations in interpretable ways.

Unfortunately, as we will see, it does not appear that a complete characterization 
