%!TEX root = ../thesis.tex
\chapter{Background}
\label{ch:background}

\section{Biology}

\subsection{Evolutionary Processes}

\subsubsection{Horizontal Gene Transfer}

\subsection{Mathematical Models of Evolution}
\label{background:ss:evolutionary_models}

\subsubsection{The Wright-Fisher Model}

\subsubsection{The Coalescent Process}
\label{background:sss:coalescent}

\subsubsection{Metrics on Sequences}

\subsection{Phylogenetic Reconstruction}

\subsubsection{Distance Matrix Methods}

\subsubsection{Phylogenetic Networks}

\subsubsection{Space of Phylogenetic Trees}

\subsubsection{Number of Trees}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% TOPOLOGICAL DATA ANALYSIS %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Topological Data Analysis}
\label{background:sec:tda}

Topology is the branch of mathematics that aims to characterize spaces up to deformations.
If we take a tree and change the branch lengths, the tree remains a tree.
How can we addess if two spaces have the same topology?
One trick is to assign to each space some algebraic object (a number, for instance) that does not change under deformation.
That object is called a topological invariant.
For instance, we can assign to an object the number of loops, which in the case of the mug will be one, and in the case of the tree will be zero.
We can only continuously deform spaces with the same invariants.
Algebraic topology provides tools to compute invariants of these spaces.

Real data does not come in the form of perfect continuous spaces, as dealt with by classical topology.
Instead, data can be viewed as a high-dimensional point cloud forming a discrete representation of a space.
Topological data analysis (TDA) refers to a framework that has been developed in the last 15 years to compute topological properties from finite point clouds, building on developments in computational topology and statistics.
Our primary tool is persistent homology, a branch of TDA that computes topological invariants representing multi-scale information about the connectivity and holes in a dataset.

\subsection{Intuition}

Topological Data Analysis (TDA) studies structure in high-dimensional data such as connectedness and the presence of holes.
In practice, we observe only a sample of data points, from which we wish to infer an underlying model or generating principle.
TDA first builds topological complexes from data, then measures informative properties of these complexes. 
Persistent homology (PH) is a method from TDA that uses algebraic topology to compute quantitative properties in data, including connectedness and the presence of holes.
For excellent review of topological data analysis, see the reviews [Carlsson, Ghrist].

Shape information is indexed by dimension.
$H_0$ information tells us about connected components and is roughly equivalent to a hierarchical clustering.
Higher dimensions represent loops ($H_1$), voids ($H_2$), and their generalizations in the data, giving a quantitative representation of shape.
The topological invariants can be concisely represented ina barcode diagram, which tracks the shape across multiple scale parameters.
Each horizontal line in the diagram represents a topological feature.

\kje{[Program: encode data as simplicial complex, combinatorial version of a topological space.
Properties studied from combinatorial, topological, algebraic perspective.]}

\subsection{Mathematical Preliminaries}

Topology: characterize properties of spaces invariant under continuous deformation.
Our goal in this section is to get to the point of defining homology.

In this section we give background sufficient to define homology for our purposes, before moving on to applied topology and persistent homology.
A more thorough exposition of algebraic topology can be found in \cite{Hatcher:2002ut}.

Associate a collection of algebraic objects with a topological space.
Quantify global properties of space.
Homotopy and Homology.
Homology: properties of chains composed of oriented simplices
Elements of homology groups are cycles (chains with vanishing boundary).
Two k-cycles are homologous if they differ by the boundary of a (k+1)-chain.
Incidence matrix representation...

We provide sufficient background to build a working definition of homology.

\subsubsection{Simplicial Complexes}

\paragraph{Simplex}

A simplex is something like a point, a line, a triangle, or a higher dimensional generalization.
Simplex: generalization of triangle or tetrahedron to arbitraty dimensions. [see Zomorodian].
k-simplex: k-dimensional polytope which is the convex hull of k+1 vertices.
Faces. Boundary.
See Example.

\begin{figure}
\centering
% \includegraphics[]{./fig/simplices.pdf}
\caption[Simplices]{}
\label{background:fig:simplices}
\end{figure}

\paragraph{Simplicial Complex}

Glue together simplices such that the following holds:

\begin{enumerate}
\item Any face of a simplex in $K$ is also in $K$
\item The intersection of any two simplices in $K$ is a face of both 
\end{enumerate}

\begin{figure}
\centering
% \includegraphics[]{./fig/simplicial_complex.pdf}
\caption[Simplicial Complex]{}
\label{background:fig:simplicial_complex}
\end{figure}

\paragraph{Chains, cycles and boundaries}

Boundary operator $\partial_{k}:C_{k}\rightarrow C_{k-1}$.
Action of boundary operator on a simplex $\sigma$ is defined as: 

\begin{equation}
\partial_{k}\sigma = \displaystyle\sum_{i}(-1)^{i}[v_{0},v_{1},...,\hat{v}_{i},...,v_{n}].
\end{equation}

A chain $C\in C_{k}$ is called a cycle if $\partial_{k}C=0$.
Chain with empty boundary.
Set of cycles forms a group.

Boundary operator defines a chain complex $C_{*}$:

\begin{equation}
\dots \overset{\partial_{n+1}}{\longrightarrow} C_n \overset{\partial_{n}}{\longrightarrow} C_{n-1} \overset{\partial_{n-1}}{\longrightarrow}  ... \overset{\partial_{2}}{\longrightarrow} C_n \overset{\partial_{1}}{\longrightarrow} C_{n-1} \overset{\partial_{0}}{\longrightarrow}  0
\end{equation}

Important property:

\begin{equation}
\partial_{k-1}\partial_{k} = 0 \forall k
\end{equation}

Intuitively, a boundary has no boundary.

$C$ is the set of 
The $k$-th cycle group is $Z_{k}=\ker \partial_{k}$.
$Z_{k}$ defines the set of all cycles of dimension $k$.
$B_{k}$ defines the set of all boundaries of dimension $k$.
That is, elements of $B_{k}$ serve as boundaries of $(k+1)$-chains.

TDA+PH: number and type of holes. which holes are essential and which are unimportant.

Get to the point where can define homology.
Chain complex.
Boundary operators
Work only over $\mod 2$ Homology (0,1) coefficients.
Torsion observed in the image patch data set, but no reason to think it is present biologial data sets we examine.

\begin{figure}
\centering
% \includegraphics[]{./fig/boundary_example.pdf}
\caption[]{}
\label{background:fig:boundary_example}
\end{figure}

\subsubsection{Homology}

Abelian groups generated by holes in the space.
Betti number is the rank of the homology group.

Simplical homology.
Abelian groups.
First homology group: abelianization of the fundamental gorup.
Quotient group

Recalling out definition of the boundary and cycle groups, define a quotient group

\begin{equation}
H_{k} = Z_{k} / B_{k} = \ker\partial_{k} / \im\partial_{k+1}
\end{equation}

Closed chains (cycles) up to boundary of higher dimensional cycles.
Elements are classes of homologous cycles.

\begin{figure}
\centering\includegraphics[width=\columnwidth]{./fig/FASY_chain_complexes.pdf}
\caption{Relationship between chain and cycle and homology. Adapted (``Adapted'') from Fasy.}
\label{fig:chain_complexes}
\end{figure}


\subsection{Constructing Spaces from Real Data}

In practice we have real data which comes to us as points in a high-dimensional space.
With a metric we can represent this as a finite metric space.
How can we apply the ideas of topology to these spaces?

\paragraph{Finite Metric Spaces}

Metric space with a finite number of points.
In topological data analysis, our spaces of interest are finite metric spaces.

\subsubsection{The \Cech\ and Vietoris-Rips Complexes}

The \Cech\ complex consists of the set of simplices $\sigma$ with vertices $v_{1},...,v_{k}\in S$ such that

\begin{equation}
\displaystyle\cap_{i}^{} B(v_{i},\epsilon)\neq0
\end{equation}

The Vietoris-Rips complex is defined as

\begin{equation}
\mathrm{VietorisRips}(r) = \{ \sigma\in S\:|\:\mathrm{diam}(\sigma) \leq 2r \}
\end{equation}
where $\mathrm{diam}(\sigma)=\{ \sup d(i,j) \:|\: i,j\in\sigma \}$

\subsubsection{Filtrations}

A series of inclusions.

\subsection{Condensed Representations}

\subsubsection{The Mapper Algorithm}
\label{subsec:mapper}

Condensed Representations.
Exploratory data analysis seeks to represent high-dimensional datasets in .

\begin{figure}
\begin{tikzpicture}[node/.style={minimum height=.8cm,minimum width=2cm,draw}]
\draw (0,0) node[node, align=center] (dr) {Dimensionality\\Reduction};
\draw (3,1) node[node] (linear) {Linear};
\draw (3,-1) node[node] (nonlinear) {Nonlinear};
\draw (6,1) node[node] (pca) {PCA};
\draw (6, -.5) node[node] (geometry) {Geometry};
\draw (6, -1.5) node[node] (topology) {Topology};
\draw (9, -.5) node[node,align=left] (manifold_learning) {Manifold\\Learning};
\draw (9, -1.5) node[node] (mapper) {Mapper};
\draw (12, -.5) node[node,align=center] (mds) {MDS\\PCA\\Isomap};


\myline{dr}{linear};
\myline{dr}{nonlinear};
\draw[-latex] (linear) -- (pca);
\myline{nonlinear}{geometry};
\myline{nonlinear}{topology};
\draw[-latex] (topology) -- (mapper);
\draw[-latex] (geometry) -- (manifold_learning);
\draw[-latex] (manifold_learning) -- (mds);

\end{tikzpicture}
\caption[Dimensionality Reduction for EDA]{Dimensionality Reduction for EDA}
\label{background:fig:eda}
\end{figure}

Mapper algorithm was first developed in \cite{Singh:2007ve}.
Mapper is coordinate free and depends onnly on the similarity of points as measured by the distance function.
Further exposition can be found in \cite{Lum:2013cz}.
Used in biology example in \cite{Nicolau:2011} top classify breast cancer subtypes.
In our work we use the commercial Mapper implementation Ayasdi \cite{AyasdiIris:2015}.
An open-source implementation of the Mapper algorithm is available in the Python Mapper package \cite{Mullner:2013}.

Steps:
(1) Project using filter function.
(2) Create overlapping bins
(3) Cluster in the projected space.
(3) Connect pairs of bins with shared points

\subsection{Persistent Homology}
\label{background:ss:persistent_homology}

Persistent refers to capturing structure that extends over multiple spatial resolutions and is in some sense robust, while homology refers to the type of topological structure that is being computed.

How to extend homology to finite metric spaces?
Data -> Sets of complexes -> Vector spaces

We summarize persistent homology from the perspective of an end-user.
For detailed background, see the reviews \cite{Carlsson:2009a,Ghrist:2008} and the books \cite{Edelsbrunner:2010,Zomorodian:2005b}.
In brief, persistent homology computes topological invariants representing information about the connectivity and holes in a dataset.
A dataset, $S=(s_{1},\ldots,s_{N})$, is represented as a point cloud in a high-dimensional space (not necessarily Euclidean).
From the point cloud, a nested family of simplicial complexes, or a filtration, is constructed, parameterized by a filtration value $\epsilon$, which controls the simplices present in the complex.
The two most common ways of constructing a simplicial complex at each $\epsilon$ are the \Cech complex and the Vietoris-Rips complex.
The filtration is represented as a list of simplices defined on the vertices of $S$, annotated with the $\epsilon$ at which the simplex appears.
Given a filtration, the persistence algorithm is used to compute homology groups.
The $0$-dimensional homology ($H_0$) represents a hierarchical clustering of the data.
Higher dimensional homology groups represent loops, holes, and higher dimensional voids in the data.
Each feature is annotated with an interval, representing the $\epsilon$ at which the feature appears and the $\epsilon$ at which the feature contracts in the filtration.
These filtration values are the \emph{birth} and \emph{death} times, respectively.

The topological invariants in the filtration can be concisely represented in a barcode diagram, a set of line segments ordered by filtration value on the horizontal axis.
In the barcode diagram,

The topological invariants in the filtration can be concisely represented in a barcode diagram, a set of line segments ordered by filtration value on the horizontal axis (Figure XXX).

Invariants can be equivalently represented by a persistence diagram, a scatter plot with the birth time on the horizontal axis and the death time on the vertical axis.

The intuition behind persistent homology is that is that good or interesting features will persist over longer scales
That is, they will be more robust.
In the barcode diagram this corresonds to long bars, and in the persistence diagram, this corresponds to points far from the diagonal.
Invariants that persist for only short scales are likely to be noise or artifacts of incomplete sampling.
The questions of how to rigorously determine what makes a good interval is an open question that is currently being addressed by a number of different groups.
We discuss this further in Section \ref{subsubsec:ph_statistics}.

\begin{figure}
\centering
\includegraphics[]{./fig/LESNICK_ExpandingBalls.pdf}
\caption[A filtration]{Example of constructing a filtration.}
\label{background:fig:exanding balls}
\end{figure}

\begin{figure}
\centering
\includegraphics[]{./fig/persistence_pipeline.pdf}
\caption[The Persistence Pipeline]{The Persistence Pipeline.}
\label{background:fig:persistence_pipeline}
\end{figure}

\subsubsection{The Persistence Algorithm}
\label{subsubsec:ph_algorithm}

While we act mostly as an end-user of persistent homology in this thesis, the algorithmics behind efficient computations of homology are interesting and worth including for comprehensiveness.
Computing persistent homology is an exercise in linear algebra.
The initial algorithms first induce a matching on a set of simplices.
This is due to Zomorodian and is the implementation used in Javaplex and Dionysus.
Then you reduce a couple of matrices.
You can read off each bar and its represetnative cycle from looking at the zeros on this one particular matrix.
Smith normal form.

More advanced algorithms have been developed that compute simplicial collapses: recognizing that the size of the simplciial set is often the limiting factor here, they collapse simplicies into simpler structures that will have identical homology.
This uses Discrete Morse Theory and is the idea behind implementations such as Perseus.

Include only simple implementation for $Z_2$.

Several packages for computing persistent homology have been developed [Dionysus, Javaplex, Guidi, phom] and TDA frontend for R.
Persistent homology is computed using Dionysus \cite{Morozov:2012}.

\subsubsection{Stability}
\label{subsubsec:ph_stability}

An important aspect of peristent homology is stability.
Stability refers to how the output of persistent homology will change when the original data is perturbed, for example due to noise or sampling.
Will the existing bars change?
Will new homology classes be formed?
We would like the output of persistent homology to be stable under these perturbations.
In general, our question is if I have some perturbation that takes my data from $D\rightarrow D'$, what can I say about the subsequent change in barcodes $B\rightarrow B'$?
In general, if I have data $D$ that is perturbed to new data $D'$, how will change
Luckily, there is a result that bounds changes in the diagram, due to Chazal and coauthors \citep{Chazal:2009wc}.
After a few definitions, we state the stability theorem.
First, we consider metrics on spaces.

\begin{defn}
\label{defn:hausdorff}
The \emph{Hausdorff distance} measures the distance between two shapes.
\begin{equation}
d_{H}(X,Y) = \max\left\{ \sup_{x \in X} \inf_{y \in Y} d(x,y), \sup_{y \in Y} \inf_{x \in X} d(X,Y) \right\}
\end{equation}
And $d_H(X,Y)=0$ iff $X=Y$.
\end{defn}

\begin{defn}
\label{defn:gromovhausdorff}
The \emph{Gromov-Hausdorff distance} measures how far two spaces are from being isometric.
It measures the longest distance from a point in one set to the closest point in another set within a metric space.
\begin{equation}
d_{GH}(X,Y)=\inf_{f,s} d_H(X,Y)
\end{equation}
\end{defn}

Next, we consider how to define the distance between two persistance diagrams.
To do so, we first need the concept of a \emph{matching}.
For two persistence diagrams $A$ and $B$, a matching is a mapping from intervals in $A$ to intervals in $B$, where we allow points to match to the diagonal to account for cases with unequal number of points.
For each matched pair of intervals $(a,b)$, we define the $L_{\infty}$ distance as

\begin{equation}
d_{\infty}(a,b) = \max\{ |a_{x}-b_{x}|, |a_{y}-b_{y}| \}.
\end{equation}


\begin{defn}
\label{defn:bottleneck}
The \emph{bottleneck cost} of a matching between two diagrams is the maximum $L_{\infty}$ for all matched points. The \emph{bottleneck distance} is defined to be the minimal bottleneck cost across all matchings. The matching with minimal bottleneck cost is the \emph{bottleneck matching}.
\begin{equation}
d_{B}(A,B) = \inf_{n:A\rightarrow B} \sup_{x\in X} ||x-\eta(x)||_{\infty}
\end{equation}
\end{defn}

The result of \citet{Chazal:2009wc} states that the bottleneck distance between $B$ and $B'$ is bounded by the Gromov-Hausdorff distance between the finite metric spaces embedded in $A$ and $B$.

\begin{thm}
\label{thm:stability}
The stability theorem.
\begin{equation}
d_{B}(H_{K}(X),H_{K}(Y)) \leq d_{GH}(X,Y)
\end{equation}
\end{thm}

This bound establishes that small perturbations in the data will produce only small changes in the the persistence diagram.

\subsubsection{Statistical Persistent Homology}
\label{subsubsec:ph_statistics}

In persistent homology, the intuition is developed that long intervals are to be interpreted as large-scale, or robust, geometric features in data, while short intervals are more likely to correspond to noise or incomplete sampling.

More broadly, how can the information in the persistent diagram be used.
Can this statement be made more precise?
How short is short, and how will noisy sampling effect the observed diagram?
When can a long interval be interpreted as a real feature, and can we assign measures of confidence to our estimates?
Substantial recent work in the TDA community has focused on these questions, in order to develop statistical foundations for persistent homology.
We give here a brief flavor of some of these ideas and their relation to our own work.

Fasy and coauthors have developed ways of generating confidence intervals for persistence diagrams \citep{Fasy:2014}.
Based on some information about density, they can put a line off the diagonal below which points are to be considered noise (see example).
Bubenik has developed the language of persistence landscapes \cite{Bubenik:2007ux,Bubenik2015:um}.  
Several authors have examined the space of persistence diagrams as a Polish space, with notions of mean and variance.
XXX et al have used the bootstrap to get estimates of the diagram robustness.
Mukherjee et al.

Also see the work of Turner \cite{Turner:2012wb} and Mileyko \cite{Mileyko:2011jm}.

\begin{itemize}
\item Probability measures on the space of persistence diagrams
\item Functional summaries of the persistence diagram
\item Confidence intervals on the persistence diagram
\item Statisical inference using persistence diagrams
\end{itemize}

Wasserstein distance between diagrams.

\begin{equation}
W_{p}(D_{1},D_{2}) = \left( \inf_{\gamma} \sum_{x\in D_1} \|x-\gamma(x)\|^{p}_{\infty} \right)^{1/p}
\end{equation}

\kje{[Expand substantially, this section needs to be very strong.]}

\subsubsection{Multidimensional Persistence}
\label{subsubsec:ph_multidim}

First laid out in \cite{Carlsson:2009b}.
More work in \cite{Lesnick:2012uv}.
Filtrations along different dimensions; how to relate?.
Prototypical example: density and distance.

Our case is going to be slightly different.
We will consider a set of points annotated with different metrics that we can put on it which will induce different homologies.
Then we will see what happens we interpolate between those different metrics.
\kje{[Discuss with Michael.]}

\section{Applying TDA to Molecular Sequence Data}

Aligned genomic sequences can be naturally viewed as points in a high-dimensional sequence space.
As more genomes continue to be sequenced, this space becomes more densely sampled.
Using the standard genetic distance metrics described above, we can compute a pairwise distance matrix betwee genomes.
This defines our finite metric space.
From there, methods from TDA such as persistent homology and mapper can be applied.
Phylogenetic information can be read off the resulting topological structures.

An important foundational point was described in \cite{Chan:2013}.
In that paper, it was shown that if the evolutionary history contained in a particular data-set is tree-like, then there will be no higher homology in the resulting barcode diagram.
In other words, the only nontrivial topology will appear in dimension zero.

If the set of genomes permits a phylogenetic representation, then positive-dimensional Betti numbers should vanish, since the topology of a tree is contractible. 
\kje{[More about Gunnar's proof?]}
If the evolutionary history includes reticulate events that cannot be represented as a tree, these events will be captured as non-trivial higher dimensional homology in the barcode diagram.

We illustrate a simple example of how TDA can capture horizontal evolution from population data in Figure~\ref{background:fig:tda_on_sequencedata}.
Consider the reticulate phylogeny (Figure~\ref{background:fig:tda_on_sequencedata}A): five genetic sequences sampled today (yellow circles) originate from a single common ancestor due to clonal evolution (solid blue lines tracing parent to offspring) and reticulate evolution (dotted red lines).
In Figure~\ref{background:fig:tda_on_sequencedata}B, these five samples are placed in the context of a larger dataset, where the data has been projected onto the plane using PCA.
Persistent homology is then applied to this larger sample.
In Figure~\ref{background:fig:tda_on_sequencedata}C we demonstrate the construction of a filtered simplicial complex, showing how the connectivity changes as the scale parameter $\epsilon$ is increased.
Finally, in Figure~\ref{background:fig:tda_on_sequencedata}D we see the resulting barcode diagram.
Using $H_0$ we can track the number of strains or subclades that persist, roughly corresponding to the tree-like component of the data.
The $H_1$ bar near spanning roughly $\epsilon=0.13$ to $\epsilon=0.16$ identifies the presence of a reticulate event. involving the five higlighted sequences.
The scale over which this bar persists represents the amount of evolutionary time separating the parents and the reticulate offspring.
Additionally, the persistence algorithm will return a generating basis for a particular homology group, which we can use to identify the particular mixtures of sequences involved a reticulation.
In this way, we can analyze both the scale and frequency of reticulation in genomic data sets.

We summarize the connection between genomics and TDA in Table~\ref{table:dictionary}.

\begin{table}
\centering
\caption{Dictionary connecting algebraic topology and evolutionary biology}
\small
\rowcolors{2}{gray!25}{white}
\setlength{\aboverulesep}{0pt}
\setlength{\belowrulesep}{0pt}
\setlength{\extrarowheight}{.75ex}
\begin{tabularx}{\textwidth}{XX}
\toprule\rowcolor{gray!50}
Algebraic Topology                                         & Evolutionary Biology \\
\midrule
Filtration value $\epsilon$                                & Genetic distance (evolutionary scale) \\
0-dimensional Betti number at filtration value $\epsilon$  & Number of clusters at scale $\epsilon$ \\
Generators of 0-D homology                                 & A representative element of the cluster \\
Hierarchical relationship among generators of 0-D homology & Hierarchical clustering \\
1-D Betti number                                           & Lower bound on number of reticulate events \\
Generators of 1-D Homology                                 & Reticulate events \\
Generators of 2-D Homology                                 & Complex horizontal genomic exchange \\
Non-zero high-dimensional homology (“topological obstruction to phylogeny”) & No treelike phylogenetic representation exists \\
Number of higher-dimensional generators over a time interval (“irreducible cycle rate”) & Lower bound on recombination/reassortment rate \\
\bottomrule
\end{tabularx}
\label{table:dictionary}
\end{table}

\begin{figure}
\centering
\includegraphics[width=\textwidth]{./fig/tda_on_sequencedata.pdf}
\caption[Applying TDA to Sequence Data]{Applying persistent homology to genomic data. (A) An evolutionary genealogy including reticulation. (B) Data projected into 2-dimensions. (C) Construction of a filtered simplicial complex. (D) The resulting multiscale barcode diagram.}
\label{background:fig:tda_on_sequencedata}
\end{figure}

\subsection{The Four Gamete Test: The Simplest Example}

In population genetics and phylogenetics, the four gamete test is a simple method for testing for the presence of reticulate evolution.
In any given dataset, the simultaneous presence of haplotype patterns 00, 10, 01, and 11 is incompatible with strictly clonal evolution.
\kje{[Needs more explanation why?]}

Using a Hamming metric, we have a finite metric space which describes a simple diamond.
The barcode diagram will reflect nontrivial topology in the interval $[1,2)$.
This simple example serves as motivation for much of the work that follows.

\subsection{The Space of Trees, Revisited}

In Section~XX we introduced the tree space as a subspace of the space of finite metric spaces.
Real data will not often sit in tree space, and the goal of phylogenetics can be seen as finding the best tree representation of data via some projection onto tree space.
Our program can be interpreted in a similar fashion: data comes to us as a finite metric space.
Rather than projecting onto tree space, we want to characterize it as it is, by defining topological invariants which characterize the deviation from tree-like additivity.
The further the data sits from tree space, the more reticulation we expect to find.